core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -180,6 +180,11 @@ public FileScanTask next() {
       this.fileScanTask = fileScanTask;
     }
 
+    @Override
+    public RowFilter createRowFilter(FileFormat format) {
```

Comment:
Does Predicate function changes per file?

Reply:
Why don't we instantiate the predicate once instead of per file?

Reply:
Yes, the actual row filter is different between files. Each file will contain different segments (opal concepts), and each segment will have its own bitmap.